Snowflake.Data/Core/RestRequestImpl.cs

```diff
@@ -94,36 +89,26 @@ public JObject get(SFRestRequest getRequest)
             message.Headers.Accept.Add(applicationSnowflake);
             message.Properties["TIMEOUT_PER_HTTP_REQUEST"] = getRequest.httpRequestTimeout;
 
-            CancellationTokenSource cancellationTokenSource = 
-                new CancellationTokenSource(getRequest.sfRestRequestTimeout);
-
-            var responseContent = sendRequest(message, cancellationTokenSource.Token).Content;
+            var responseContent = sendRequest(message, getRequest.sfRestRequestTimeout).Content;
 
             var jsonString = responseContent.ReadAsStringAsync();
             jsonString.Wait();
 
             return JObject.Parse(jsonString.Result);
         }
 
-        private HttpResponseMessage sendRequest(HttpRequestMessage requestMessage, CancellationToken cancellationToken)
+        private HttpResponseMessage sendRequest(HttpRequestMessage requestMessage, TimeSpan timeout)
         {
             try
             {
-                var response = HttpUtil.getHttpClient().SendAsync(requestMessage, cancellationToken)
+                var response = HttpUtil.initHttpClient(timeout).SendAsync(requestMessage)

```

Comment:
The timeout here is passed down to the constructor of http client and set to Timeout property of HttpClient. https://msdn.microsoft.com/en-us/library/system.net.http.httpclient.timeout(v=vs.110).aspx
According to the doc, this timeout looks to me is still the per http request, (per http request timeout is passed down using a map HttpRequestMessage.Properties) This timeout is the total timeout when accessing a snowflake rest endpoint (including all the retries). So I think a cancellation token will make more sense here.